### PR TITLE
docs(framework) Update how to use strategies guide

### PR DESCRIPTION
**Pull request by adam-narozniak**
Update to use the up-to-date strategy-reated API.

### CODE REVIEW

File: doc/source/how-to-use-strategies.rst [hunk 1]
```diff
@@ -14,49 +14,59 @@ the server side:
...
+    app = ServerApp(server_fn=server_fn)
 
-    strategy = fl.server.strategy.FedAvg(
-        fraction_fit=0.1,  # Sample 10% of available clients for the next round
-        min_fit_clients=10,  # Minimum number of clients to be sampled for the next round
-        min_available_clients=80,  # Minimum number of clients that need to be connected to the server before a training round can start
-    )
-    fl.server.start_server(config=fl.server.ServerConfig(num_rounds=3), strategy=strategy)
+A Strategy needs to be instantiated inside a function that takes ``Context`` and returns
+``ServerAppComponents`` (e.g. named ``server_fn`` in this example). The parameters of
```
Comment (chongshenng) [inline doc/source/how-to-use-strategies.rst:43]:
```suggestion
A Strategy needs to be instantiated inside a function (e.g. ``server_fn`` in this example) that takes ``Context`` and returns a
``ServerAppComponents`` object. The parameters of
```

File: doc/source/how-to-use-strategies.rst [hunk 2]
```diff
@@ -14,49 +14,59 @@ the server side:
...
+the strategy can be customized by accessing them from the context or or hard-coding
+directly in the function. The ``server_fn`` function needs to be passed to a
```
Comment (chongshenng) [inline doc/source/how-to-use-strategies.rst:45]:
```suggestion
the strategy can be customized by accessing them from the context, which in turn
are set under the ``[tool.flwr.app.config]`` table in your ``pyproject.toml``. The ``server_fn`` function needs to be passed to a
```

File: doc/source/how-to-use-strategies.rst [hunk 3]
```diff
@@ -73,18 +83,28 @@ client. It must return a dictionary of arbitrary configuration values ``client.f
...
+    app = ServerApp(server_fn=server_fn)
 
 The ``on_fit_config_fn`` can be used to pass arbitrary configuration values from server
-to client, and potentially change these values each round, for example, to adjust the
+to client and potentially change these values each round, for example, to adjust the
 learning rate. The client will receive the dictionary returned by the
-``on_fit_config_fn`` in its own ``client.fit()`` function.
+``on_fit_config_fn`` in its own ``client.fit()`` function. And while the values can be
+also passed directly form via the context this function can be a place of adding some
+randomness that can't be achieved by the context, which is static.
```
Comment (chongshenng) [inline doc/source/how-to-use-strategies.rst:107]:
```suggestion
also passed directly via the context this function can be a place to implement
finer control over the `fit` behaviour that may not be achieved by the context, which sets fixed values.
```

**Review by chongshenng — COMMENTED** [review]
Thanks for the update @adam-narozniak! Some small suggestions for rewording. Otherwise it LGTM!

File: doc/source/how-to-use-strategies.rst [hunk 4]
```diff
@@ -14,49 +14,68 @@ the server side:
...
 
-This creates a strategy with all parameters left at their default values and passes it
-to the ``start_server`` function. It is usually recommended to adjust a few parameters
-during instantiation:
+    def server_fn(context: Context):
+        # Optional context-based parameters specification
+        num_rounds = context.run_config["num-server-rounds"]
+        config = ServerConfig(num_rounds=num_rounds)
+
+        strategy = FedAvg(
```
Comment (danieljanes) [inline doc/source/how-to-use-strategies.rst:32]:
```suggestion

        # Instantiate FedAvg strategy
        strategy = FedAvg(
```

File: doc/source/how-to-use-strategies.rst [hunk 5]
```diff
@@ -14,49 +14,68 @@ the server side:
...
+            fraction_fit=context.run_config["fraction-fit"],
+            fraction_evaluate=1.0,
+        )
+        return ServerAppComponents(strategy=strategy, config=config)
```
Comment (danieljanes) [inline doc/source/how-to-use-strategies.rst:36]:
```suggestion
        )

        return ServerAppComponents(strategy=strategy, config=config)
```

File: doc/source/how-to-use-strategies.rst [hunk 6]
```diff
@@ -14,49 +14,70 @@ the server side:
...
+object, which in turn contains the values specified in the ``[tool.flwr.app.config]``
+table in your ``pyproject.toml`` (a snippet is shown below).
 
 .. code-block:: python
 
-    import flwr as fl
+    ...
+    [tool.flwr.app.config]
+    num - server - rounds = 10
+    fraction - fit = 0.5
```
Comment (danieljanes) [inline doc/source/how-to-use-strategies.rst:55]:
```suggestion
    num-server-rounds = 10
    fraction-fit = 0.5
```

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**